rpgpy/data.pyx

```diff
@@ -144,20 +144,24 @@ def _read_rpg_l0(file_name, header):
         fseek(ptr, n_dummy * 4, SEEK_CUR)
         fread(is_data, 1, n_levels, ptr)
 
+        chirp_of_level = np.digitize(range(n_levels), header['RngOffs'])
+
         for alt_ind in range(n_levels):
 
             if is_data[alt_ind] == 1:                
                 fseek(ptr, 4, SEEK_CUR)
+                n_bins = header["SpecN"][chirp_of_level[alt_ind]-1]
 
                 if compression == 0:
 
+                    bins_to_shift = (n_spectra - n_bins)//2
```

Comment:
I think you have to calculate `bins_to_shift` before `if compression == 0:` condition.

Reply:
you are absolutely right!